src/platform/errors/errorHandler.node.ts

```diff
@@ -151,12 +156,22 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
         kernelConnection: KernelConnectionMetadata,
         resource: Resource
     ): Promise<KernelInterpreterDependencyResponse> {
-        traceWarning('Kernel Error', err);
+        traceWarning(`Kernel Error, context = ${context}`, err);
         err = WrappedError.unwrap(err);
 
         // Jupyter kernels, non zmq actually do the dependency install themselves
         if (err instanceof JupyterKernelDependencyError) {
-            return err.reason;
+            traceWarning(`Jupyter Kernel Dependency Error, reason=${err.reason}`, err);
+            if (err.reason === KernelInterpreterDependencyResponse.uiHidden) {
```

Comment:
If we don't have the dependencies and UI was disabled, now during the error handling display the UI.